### PR TITLE
[kube-dns] fix cve and build for cse edition

### PR DESCRIPTION
**Pull request by borg-z**
## Description

kubeDns coredns
kubeDns resolvWatcher
kubeDns stsPodsHostsAppenderInitContainer
kubeDns stsPodsHostsAppenderWebhook

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: <kebab-case of a module name> | <1st level dir in the repo>
type: fix | feature | chore
summary: <ONE-LINE of what effectively changes for a user>
impact: <what to expect for users, possibly MULTI-LINE>, required if impact_level is high ↓
impact_level: default | high | low
```

### CODE REVIEW

File: modules/042-kube-dns/images/coredns/werf.inc.yaml [hunk 1]
```diff
@@ -1,32 +1,42 @@
 ---
-artifact: {{ .ModuleName }}/coredns-artifact
-from: {{ .Images.BASE_GOLANG_21_ALPINE_DEV }}
+artifact: {{ .ModuleName }}/{{ $.ImageName }}-src-artifact
+from: {{ .Images.BASE_GOLANG_23_ALPINE_DEV }}
```
Comment (RomanenkoDenys) [inline modules/042-kube-dns/images/coredns/werf.inc.yaml:3]: src-artifact

File: modules/042-kube-dns/images/coredns/werf.inc.yaml [hunk 2]
```diff
@@ -1,32 +1,42 @@
...
+    to: /patches
+    stageDependencies:
+      install:
+        - '**/*'
+shell:
+  install:
+    - mkdir -p /src
+    - cd /src
+    - git clone --depth 1 -b v1.11.3 {{ $.SOURCE_REPO }}/{{ $.ImageName }}/{{ $.ImageName }}.git .
+    - find /patches -name '*.patch' -exec git apply {} \;
```
Comment (RomanenkoDenys) [inline modules/042-kube-dns/images/coredns/werf.inc.yaml:15]:
```suggestion
    - git apply /patches/*.patch --verbose 
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/openstack'...

File: modules/042-kube-dns/images/coredns/werf.inc.yaml [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
+  - artifact: {{ .ModuleName }}/{{ $.ImageName }}-src-artifact
+    add: /src
+    to: /src
+    before: install
 mount:
   - fromPath: ~/go-pkg-cache
     to: /go/pkg
 shell:
+  beforeInstall:
+  {{- include "alpine packages proxy" . | nindent 2 }}
```
Comment (RomanenkoDenys) [inline modules/042-kube-dns/images/coredns/werf.inc.yaml:22]:
```suggestion
```

File: modules/042-kube-dns/images/sts-pods-hosts-appender-init-container/werf.inc.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+artifact: {{ .ModuleName }}/{{ $.ImageName }}-artifact
+from: {{ .Images.BASE_GOLANG_23_ALPINE }}
+import:
+  - artifact: {{ .ModuleName }}/{{ $.ImageName }}-src-artifact
+    add: /src
+    to: /src
+    before: install
+shell:
+  beforeInstall:
+  {{- include "alpine packages proxy" . | nindent 2 }}
```
Comment (RomanenkoDenys) [inline modules/042-kube-dns/images/sts-pods-hosts-appender-init-container/werf.inc.yaml:20]:
```suggestion
```

File: modules/042-kube-dns/images/sts-pods-hosts-appender-webhook/werf.inc.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+artifact: {{ .ModuleName }}/{{ $.ImageName }}-artifact
+from: {{ .Images.BASE_GOLANG_23_ALPINE }}
+import:
+  - artifact: {{ .ModuleName }}/{{ $.ImageName }}-src-artifact
+    add: /src
+    to: /src
+    before: install
+shell:
+  beforeInstall:
+  {{- include "alpine packages proxy" . | nindent 2 }}
```
Comment (RomanenkoDenys) [inline modules/042-kube-dns/images/sts-pods-hosts-appender-webhook/werf.inc.yaml:20]:
```suggestion
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

File: modules/042-kube-dns/images/coredns/werf.inc.yaml [hunk 6]
```diff
@@ -1,32 +1,37 @@
...
-  stageDependencies:
-    install:
-      - '**/*'
+artifact: {{ .ModuleName }}/{{ $.ImageName }}-src-artifact
+fromArtifact: src-artifact
+shell:
+  install:
+    - mkdir -p /src
+    - cd /src
+    - git clone --depth 1 -b v1.11.3 {{ $.SOURCE_REPO }}/{{ $.ImageName }}/{{ $.ImageName }}.git .
```
Comment (RomanenkoDenys) [inline modules/042-kube-dns/images/coredns/werf.inc.yaml:8]:
```suggestion
    - git clone --depth 1 -b v1.11.3 {{ $.SOURCE_REPO }}/{{ $.ImageName }}/{{ $.ImageName }}.git /src
```

**Review by RomanenkoDenys — APPROVED** [review]
one nit

**Pull request merged**